hazelcast/src/main/java/com/hazelcast/topic/impl/TotalOrderedTopicProxy.java

```diff
@@ -47,4 +62,53 @@ public void publish(@Nonnull  E message) {
         InternalCompletableFuture f = invokeOnPartition(operation);
         f.joinInternal();
     }
+
+    @Override
+    public InternalCompletableFuture<Void> publishAsync(@Nonnull E message) {
+        checkNotNull(message, NULL_MESSAGE_IS_NOT_ALLOWED);
+        Operation op = new PublishOperation(getName(), toData(message)).setPartitionId(partitionId);
+        return publishInternalAsync(op);
+    }
+
+    @Override
+    public void publishAll(@Nonnull Collection<? extends E> messages) {
+        checkNotNull(messages, NULL_MESSAGE_IS_NOT_ALLOWED);
+        checkNoNullInside(messages, NULL_MESSAGE_IS_NOT_ALLOWED);
+        // RU_COMPAT_4_0
+        checkClusterVersion(Versions.V4_1);
+
+        Operation op = new PublishAllOperation(getName(), toDataArray(messages));
+        publishInternalAsync(op).joinInternal();
+    }
+
+    @Override
+    public InternalCompletableFuture<Void> publishAllAsync(@Nonnull Collection<? extends E> messages) {
+        checkNotNull(messages, NULL_MESSAGE_IS_NOT_ALLOWED);
+        checkNoNullInside(messages, NULL_MESSAGE_IS_NOT_ALLOWED);
+        // RU_COMPAT_4_0
+        checkClusterVersion(Versions.V4_1);
+
+        Operation op = new PublishAllOperation(getName(), toDataArray(messages));
+        return publishInternalAsync(op);
+    }
+
+    private InternalCompletableFuture<Void> publishInternalAsync(Operation operation) {
+        topicStats.incrementPublishes();
+        try {
+            return operationService.invokeOnPartition(OperationService.SERVICE_NAME, operation, partitionId);
```

Comment:
Again, the issue with successfully published statistics. Maybe you can avoid it alltogether, and keep the current behaviour, and I'll fix it in another PR in all places.

Reply:
As mentioned earlier, I think I've found every call of `incrementPublishes`

Reply:
Ideally, we should increment once the future is done, but let's not nitpick now, I'll do it later.